tests/projects/test_tasks_caching.py

```diff
@@ -412,3 +414,125 @@ def test_new_phase_resets_cache(
             assert len(active_projects) == 3
             assert len(past_projects) == 0
             assert len(future_projects) == 0
+
+
+@pytest.mark.django_db
+def test_module_publish_resets_cache(
+    client,
+    user,
+    phase_factory,
+    project_factory,
+    django_assert_num_queries,
+    django_capture_on_commit_callbacks,
+):
+    n_objects = 3
+    objects = project_factory.create_batch(size=n_objects)
+
+    # make dates to work with phase_factory
+    now = datetime.now(tz=timezone.utc)
+    last_week = now - timedelta(days=7)
+    next_week = now + timedelta(days=7)
+
+    # make active projects
+    active_project = objects[0]
+    phase_factory(
+        start_date=last_week,
+        end_date=next_week,
+        module__project=active_project,
+    )
+
+    past_project = objects[1]
+    phase_factory(
+        start_date=last_week,
+        end_date=last_week + timedelta(days=6),
+        module__project=past_project,
+    )
+    # make future project
+    future_project = objects[2]
+    phase_factory(
+        start_date=next_week,
+        end_date=next_week + timedelta(days=7),
+        module__project=future_project,
+    )
+
+    # check function set_cache_for_projects
+    set_cache_for_projects()
+
+    # check projects got cached correctly
+    with freeze_time(now):
+        with django_assert_num_queries(0):
+            active_projects = cache.get("projects_activeParticipation")
+            past_projects = cache.get("projects_pastParticipation")
+            future_projects = cache.get("projects_futureParticipation")
+            assert len(active_projects) == 1
+            assert len(past_projects) == 1
+            assert len(future_projects) == 1
+
+    # turn future project into active
+    phase_active = phase_factory(
+        start_date=last_week,
+        end_date=next_week,
+        module__project=future_project,
+    )
+    # turn past project into future project
+    phase_future = phase_factory(
+        start_date=next_week,
+        end_date=next_week + timedelta(days=7),
+        module__project=past_project,
+    )
+
+    module1 = phase_active.module
```

Comment:
so this is a new module? didn't realise it. so every phase factory autocreates a new module? @goapunk

Reply:
unless you specify an existing module it creates a new module, yes